cmake/OpenCVDetectCUDA.cmake

```diff
@@ -28,6 +28,18 @@ endif()
 
 if(CUDA_FOUND)
   set(HAVE_CUDA 1)
+  if(CUDA_VERSION VERSION_GREATER_EQUAL "11.0")
```

Comment:
I don't think it's a good idea to hard code CUDA version here.
It may change again in future, and twisting based on version number is just increasing the chaos.
Let's filter out the library based on result of helper function.